srtcore/api.cpp

```diff
@@ -3157,7 +3157,14 @@ void srt::CUDTUnited::updateMux(CUDTSocket* s, const sockaddr_any& reqaddr, cons
         m.m_pSndQueue = new CSndQueue;
         m.m_pSndQueue->init(m.m_pChannel, m.m_pTimer);
         m.m_pRcvQueue = new CRcvQueue;
-        m.m_pRcvQueue->init(128, s->core().maxPayloadSize(), m.m_iIPversion, 1024, m.m_pChannel, m.m_pTimer);
+
+        // We can't use maxPayloadSize() because this value isn't valid until the connection is established.
+        // We need to "think big", that is, allocate a size that would fit both IPv4 and IPv6.
+        size_t payload_size = s->core().m_config.iMSS - CPacket::HDR_SIZE - CPacket::udpHeaderSize(AF_INET);
```

Comment:
There's a controversy whether this size should be used, or rather the payload size set by an option, if it's less, to allocate less memory. I think this shouldn't be done because if you happen to receive an SRT packet that would be oversized in these conditions, it will end up in a call that results in getting MSG_TRUNC flag due to too small buffer for the packet, which may lead to hard to detect errors.